src/variable.cpp

```diff
@@ -921,52 +897,52 @@ template <class T> VariableSlice VariableSlice::assign(const T &other) {
   return *this;
 }
 
-template VariableSlice VariableSlice::assign(const Variable &);
-template VariableSlice VariableSlice::assign(const ConstVariableSlice &);
+template VariableSlice VariableSlice::assign(const Variable &) const;
```

Comment:
Haven't looked into the details behind this, but there are `const` methods here I would assume to be mutable operations, `assign`, `operator+=` etc.

Reply:
Yes that may be surprising at first: The reason is that the `const`-ness of the references object (variable) is reflected in the type (`ConstVariableSlice` vs. `VariableSlice`), *not* the `const`-ness of the slice, since it is a proxy object.

You will observe the same, e.g., for `std::span`: https://en.cppreference.com/w/cpp/container/span/begin (not how both `begin()` and `cbegin()` are `const`). In this case const-correctness work by having `std::span<double>` and `std::span<const double>`.